kwant_to_openfermion.py

```diff
@@ -1,41 +1,95 @@
 import openfermion
 import kwant
+import numpy
+import tinyarray
 
+def _index(lattice_index, spin_index, n_spin):
+    '''
+    Create an integer index based on lattice and spin indices, following the openfermion convension.
+    '''
+    return lattice_index*n_spin + spin_index
+
+def _check_dimension(value, n):
+    '''
+    Check if value is a numeric array of size n x n (or a scalar in case of n=1)
+    '''
+    #if value is an array
+    if isinstance(value, (numpy.ndarray, tinyarray.ndarray_complex, tinyarray.ndarray_float, tinyarray.ndarray_int)):
+        if value.shape == (n, n):
+            return
+        raise ValueError(f'Expected an array of shape ({n},{n}), got {value.shape}')
+    #numeric value accepted only for n=1
+    if isinstance(value, (int, float, complex)):
+        if n == 1:
+            return
+        raise ValueError(f'Got a scalar, please convert to  an ({n},{n}) array')
+    #Else: wrong type
+    raise TypeError(f'Expected a number or a numeric array, got {type(value)}')
+
+
+def _single_term_to_FermionOperator(val, lat_ix1, lat_ix2, n_spin):
+    '''
+    Export single term of the hamiltonian to openfermion.
```

Comment:
I would state what a 'single term' here is (i.e. that it is a single term of the form c^{\dag}_i c_j), and mention that kwant handles the hermitian conjugate internally.